tests/tests/src/core/Core_ArrayTests.res

```diff
@@ -109,3 +109,36 @@ Test.run(
 
 Test.run(__POS_OF__("last - with items"), [1, 2, 3]->Array.last, eq, Some(3))
 Test.run(__POS_OF__("last - empty"), []->Array.last, eq, None)
+
+{
+  let array = []
+  array->Array.splice(~start=1, ~remove=0, ~insert=["foo"])
+  Test.run(__POS_OF__("splice - Insert no delete"), array, eq, ["foo"])
+}
+
+{
+  let array = ["bar", "baz"]
+  Test.run(
+    __POS_OF__("splice - Insert and delete"),
+    (array->Array.splice(~start=1, ~remove=1, ~insert=["foo"]), array),
+    eq,
+    (
+      // Even though original .splice returns an array with the removed items,
+      // the binding returns unit so there's no confusion about it mutating the original array.
+      (),
+      ["bar", "foo"],
+    ),
+  )
+}
+
+{
+  let array = []
+  array->Array.removeInPlace(0)
+  Test.run(__POS_OF__("removeInPlace - empty"), array, eq, [])
+}
+
+{
+  let array = ["Hello", "Hi", "Good bye"]
+  array->Array.removeInPlace(1)
+  Test.run(__POS_OF__("removeInPlace - from middle"), array, eq, ["Hello", "Good bye"])
+}
```

Comment:
should we keep these tests here given they're already in the docstring tests?

Reply:
Are the docstring tests ran automatically?

Reply:
@DZakh yes, that's the whole point of those docstring tests :)